### PR TITLE
refactor(codewarnings): Class memeber and local member changes (fix code warnings)

### PR DESCRIPTION
**Pull request by jamdavi**
[Class structure] Field can be local
[Declaration redundancy] Declaration can have final modifier

### CODE REVIEW

**Review by azabbasi — APPROVED** [review]

File: deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpDeviceOperations.java [hunk 1]
```diff
@@ -14,9 +14,12 @@
 
 public class AmqpDeviceOperations
 {
+    @SuppressWarnings("CanBeFinal") // Can be inherited and changed somewhere else besides the constructor
```
Comment (bikamani) [inline deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpDeviceOperations.java:17]: Is it better to suppress at class level?

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java [hunk 2]
```diff
@@ -35,7 +35,9 @@
         OPENED
     }
 
+    @SuppressWarnings("CanBeFinal") // Public member can be changed
     public static long SEND_PERIOD_MILLIS = 10L;
```
Comment (bikamani) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java:39]: How is it changed? It's static

Comment (bikamani) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/TransportClient.java:39]: I understood now.

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java [hunk 3]
```diff
@@ -239,14 +238,14 @@ else if (proxySettings.getProxy().type() == Proxy.Type.HTTP)
         this.deviceMethod = new MqttDeviceMethod(
             mqttAsyncClient,
             deviceId,
-            this.connectOptions,
+                connectOptions,
```
Comment (bikamani) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java:241]: Extra spaces?

File: provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java [hunk 4]
```diff
@@ -280,6 +280,7 @@ private TPMT_PUBLIC createPersistentPrimary(Tpm tpm, TPM_HANDLE hPersistent, TPM
         return cpResp.outPublic;
     }
 
+    @SuppressWarnings("SameParameterValue")
```
Comment (bikamani) [inline provisioning/security/tpm-provider-emulator/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/hsm/SecurityProviderTPMEmulator.java:283]: What is this warning for?

**Review by bikamani — APPROVED** [review]

File: deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpDeviceOperations.java (hunk 1 above)
Comment (jamdavi) [reply deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpDeviceOperations.java:17]: We thought about this, however we have inconsistent protection levels and if we make changes to the class with private members we want the analyzer to produce meaningful warnings in the future.

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java (hunk 3 above)
Comment (jamdavi) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/mqtt/MqttIotHubConnection.java:241]: Yeah I don't like that either. Will fix

**Pull request merged**